launcher/ui/dialogs/ProfileSetupDialog.cpp

```diff
@@ -261,7 +264,21 @@ void ProfileSetupDialog::setupProfileFinished()
     } else {
         auto parsedError = MojangError::fromJSON(*m_profile_response);
         ui->errorLabel->setVisible(true);
-        ui->errorLabel->setText(tr("The server returned the following error:") + "\n\n" + parsedError.errorMessage);
+
+        QString errorMessage = tr("Network error(%1): %2 ;HTTP Status: %3")
+                                   .arg(QString::number(m_profile_task->error()), m_profile_task->errorString(),
+                                        QString::number(m_profile_task->replyStatusCode()));
+
+        if (parsedError.fullyParsed) {
+            errorMessage += "Path: " + parsedError.path + "\n";
+            errorMessage += "Error: " + parsedError.error + "\n";
+            errorMessage += "Error message: " + parsedError.errorMessage + "\n";
+        } else {
+            errorMessage += "Failed to parse Mojang error: " + parsedError.parseError.errorString() + "\n";
+            errorMessage += "Raw contents:\n" + parsedError.rawError + "\n";
```

Comment:
```suggestion
            errorMessage += "Log:\n" + parsedError.rawError + "\n";
```
It would be cool to maybe have this in a collapse section so it's not always shown?
I don't know how much harder that would be to implement though, so up to you :)